src/Security/Handler/LogoutSuccessHandler.php

```diff
@@ -35,31 +37,50 @@ class LogoutSuccessHandler implements LogoutSuccessHandlerInterface
     /** @var AuthorizationHeaderTokenExtractor $tokenExtractor */
     private $tokenExtractor;
 
-    /** @var TokenStorage $tokenStorage */
-    private $tokenStorage;
+    /** @var CacheItemPoolInterface */
+    private $tokenCache;
+
+    /** @var JwtTokenCacheIdGenerator */
+    private $tokenCacheIdGenerator;
+
+    /** @var LoggerInterface */
+    private $logger;
 
     /** @var SerializerResponder */
     private $serializerResponder;
 
     public function __construct(
         AuthorizationHeaderTokenExtractor $tokenExtractor,
-        TokenStorage $jwtStorage,
+        CacheItemPoolInterface $jwtTokenCache,
+        JwtTokenCacheIdGenerator $tokenCacheIdGenerator,
+        LoggerInterface $securityLogger,
         SerializerResponder $serializerResponder
     ) {
         $this->tokenExtractor = $tokenExtractor;
-        $this->tokenStorage = $jwtStorage;
+        $this->tokenCache = $jwtTokenCache;
+        $this->tokenCacheIdGenerator = $tokenCacheIdGenerator;
+        $this->logger = $securityLogger;
         $this->serializerResponder = $serializerResponder;
     }
 
     public function onLogoutSuccess(Request $request)
     {
-        $credentials = $this->tokenExtractor->extract($request);
+        $accessToken = $this->tokenExtractor->extract($request);
+        $parsedToken = (new Parser())->parse($accessToken);
 
-        $token = (new Parser())->parse($credentials);
+        $refreshTokenCacheId = $this->tokenCacheIdGenerator->generate($parsedToken, 'refreshToken');
+        $cacheItem = $this->tokenCache->getItem($refreshTokenCacheId);
 
-        $username = $token->getClaim('username');
+        if ($cacheItem->isHit()) {
```

Comment:
Why don't we delete the token if isHit? Are you sure?

Reply:
What? I don't understand the question sorry. We are deleting it in the next row.

Reply:
My bad. I miss understood the isHit Method